### PR TITLE
feat(ngMock): add support for `$animate.closeAndFlush()`

### LINKED ISSUE #13005: $animate.flush() does not flush $animateCss animations in queue
We have an issue in https://github.com/angular-ui/bootstrap/pull/4493 where it turns out that `$animate.flush()` does not flush the animations queued up by $animateCss, so the `expandDone` callback in the promise chain never fires in the collapse directive, which is causing some testing problems.

In particular, the `.then` callback in the promise returned by `$animateCss(...).start()` never gets called because there is no mechanism in ngMock for flushing out the queued animation.

### PR DESCRIPTION
**Pull request by matsko**
Use `$animate.closeAndFlush()` to close all running animations.

Closes #13005

### CODE REVIEW

File: src/ngAnimate/animateCss.js [hunk 1]
```diff
@@ -456,7 +456,7 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
       var node = getDomNode(element);
       if (!node
           || !node.parentNode
-          || !$animate.enabled()) {
+          || !$$animateQueue.enabled()) {
```
Comment (gkalpak) [inline src/ngAnimate/animateCss.js]: Why this change ?

Comment (Narretz) [inline src/ngAnimate/animateCss.js]: ngMock decorates the $animate service with the new closeAndFlush method. Both $animateCss and $$animateJs have their own closeAndFlush implementations so, $animate requires both services. Since $animateCss also requires $animate, this would create a circular dependency.

File: src/ngAnimate/animateJs.js [hunk 2]
```diff
@@ -63,8 +65,32 @@ var $$AnimateJsProvider = ['$animateProvider', function($animateProvider) {
         applyAnimationClasses(element, options);
       }
 
+      function close(success) {
```
Comment (gkalpak) [inline src/ngAnimate/animateJs.js]: `success` isn't used.

**Comment by gkalpak** [comment]
LGTM (other than the unused `success` argument).
